src/main/java/com/puppycrawl/tools/checkstyle/DefaultConfiguration.java

```diff
@@ -103,8 +102,8 @@ public String getProperty(String propertyName) throws CheckstyleException {
 
     @Override
     public Configuration[] getChildren() {
-        return children.toArray(
-                EMPTY_CONFIGURATION_ARRAY);
+        final Configuration @ArrayLen(0) [] emptyArray = new Configuration[0];
```

Comment:
```suggestion
        final Configuration @ArrayLen(0)[] emptyArray = new Configuration[0];
```
I would expect `NoWhitespaceBefore`, but its not actually finding that much as figured out in #16810, to fail the build for the whitespace.